src/foremast/gcp_iam/create_iam_resources.py

```diff
@@ -0,0 +1,88 @@
+from googleapiclient.errors import HttpError
+
+from ..utils.gcp_environment import GcpEnvironment
+from . import create_service_account, get_policy, set_policy, modify_policy_remove_member, modify_policy_add_binding
+
+import logging
+import time
+
+LOG = logging.getLogger(__name__)
+
+
+def create_iam_resources(env: GcpEnvironment, app_name: str, services: dict = None):
+    credentials = env.get_credentials()
+    # Ensure service account is created
+    service_account = create_service_account(credentials=credentials,
+                                             project_id=env.service_account_project,
+                                             name=app_name)
+    member = "serviceAccount:" + service_account["email"]
+
+    # Get the gcp_roles requested in pipeline.json with the full project id as the key
+    gcp_roles_by_project = _get_gcp_roles_by_project(services['gcp_roles'], env) if "gcp_roles" in services else dict()
+
+    # For each project in this GCP environment, get the policy
+    # remove any references to this svc account, then re-add based on pipeline.json
+    # If changes were made locally, update the policy in GCP
+    for project in env.get_all_projects():
+        # Policies in GCP are shared between users/service accounts
+        # Attempt to update this project policy, if GCP returns a
+        # a 409 conflict we must re-retrieve the policy, re-update it
+        # and attempt to save again to save it again
+        attempts = 0
+        max_attempts = 5
```

Comment:
https://github.com/foremast/foremast/blob/da668786740a9247c4fb7d77f3c651ef6084dd7e/src/foremast/utils/elb.py#L28

might be better to use the existing retries decorator on a function call? Thoughts?

Reply:
Yep done, I think a backoff is necessary and thought that wasn't possible with this library but I was wrong.  Fixed!

Reply:
Awesome. Thanks. Looks much nicer.